packages/flame/lib/src/geometry/ray2.dart

```diff
@@ -0,0 +1,101 @@
+import 'dart:math';
+
+import 'package:flame/geometry.dart';
+import 'package:meta/meta.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+/// A ray in the 2d plane.
+///
+/// The [direction] should be normalized.
+class Ray2 {
+  Ray2(this.origin, Vector2 direction) {
+    this.direction = direction;
+  }
+
+  Ray2.empty() : this(Vector2.zero(), Vector2(1, 0));
+
+  Vector2 origin;
+  late Vector2 _direction;
+  Vector2 get direction => _direction;
+  set direction(Vector2 direction) {
+    assert(
+      direction.x.abs() <= 1 && direction.y.abs() <= 1,
+      'direction must be normalized',
+    );
+    _direction = direction;
```

Comment:
```suggestion
    _direction.setFrom(direction);
```
then the `_direction` can be final

Reply:
The problem is that then we need to create another vector for the initialization of the ray, but maybe that is worth it.